pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -874,6 +937,65 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 	return claim, nil
 }
 
+// isClaimBound checks whether a given resource claim is successfully
+// bound to a device.
+func (pl *DynamicResources) isClaimBound(claim *resourceapi.ResourceClaim) (bool, error) {
+	for _, deviceRequest := range claim.Status.Allocation.Devices.Results {
+		if len(deviceRequest.BindingConditions) == 0 {
+			continue
+		}
+		deviceStatus := getAllocatedDeviceStatus(claim, &deviceRequest)
+		if deviceStatus == nil {
+			return false, nil
+		}
+		for _, cond := range deviceRequest.BindingFailureConditions {
+			if apimeta.IsStatusConditionTrue(deviceStatus.Conditions, cond) {
+				return false, fmt.Errorf("claim %s failed to bind", claim.Name)
+			}
+		}
+		for _, cond := range deviceRequest.BindingConditions {
+			if !apimeta.IsStatusConditionTrue(deviceStatus.Conditions, cond) {
+				return false, nil
+			}
+		}
+	}
+	return true, nil
+}
+
+// hasDeviceBindingStatus checks the binding status of devices within the
+// given state claims.
+func (pl *DynamicResources) hasDeviceBindingStatus(ctx context.Context, state *stateData, pod *v1.Pod, nodeName string) (bool, error) {
+	for claimIndex, claim := range state.claims {
+		claim, err := pl.clientset.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		state.claims[claimIndex] = claim
+		bound, err := pl.isClaimBound(claim)
+		if err != nil {
+			return false, err
+		}
+		if !bound {
+			pl.fh.EventRecorder().Eventf(claim, pod, v1.EventTypeNormal, "BindingConditionsPending", "Scheduling", "waiting for binding conditions for device on node %s.", nodeName)
```

Comment:
This gets emitted every five second (polling interval in wait.PollUntilContextTimeout). That seems a bit excessive.

Wouldn't it be enough to emit once, when starting to wait?

Reply:
Changes the Event to be emitted only when the wait begins.